Enable push notifications by default

cammellos opened the pull request:
depends on #2022 

This PR does a few things:
- Enable sending push notifications by default
- Process push notifications at-most-once on the server side
- Filter by chatID push notifications on server side
- Set default push notifications servers when starting the app

cammellos committed:
Enabled push notifications by default and add fleet

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (76)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 07cedd2c | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/1/) | 2020-08-20 09:13:04 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200820-091013-07cedd.zip) |
| :heavy_check_mark: | 07cedd2c | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/1/) | 2020-08-20 09:13:37 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200820-091011-07cedd.zip) |
| :heavy_check_mark: | 07cedd2c | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/1/) | 2020-08-20 09:17:48 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200820-091009-07cedd.aar) |
| | | | | | | |
| :heavy_check_mark: | 49447698 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/2/) | 2020-08-20 10:10:26 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200820-100832-494476.zip) |
| :heavy_check_mark: | 49447698 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/2/) | 2020-08-20 10:10:38 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200820-100830-494476.zip) |
| :heavy_check_mark: | 49447698 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/2/) | 2020-08-20 10:14:14 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200820-100828-494476.aar) |
| | | | | | | |
| :heavy_check_mark: | 0e8cba34 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/3/) | 2020-08-20 14:00:44 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200820-135902-0e8cba.zip) |
| :heavy_check_mark: | 0e8cba34 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/3/) | 2020-08-20 14:01:10 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200820-135904-0e8cba.zip) |
| :heavy_check_mark: | 0e8cba34 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/3/) | 2020-08-20 14:05:20 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200820-135900-0e8cba.aar) |
| | | | | | | |
| :heavy_check_mark: | ea8e0d49 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/4/) | 2020-08-24 08:52:53 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-085058-ea8e0d.zip) |
| :heavy_check_mark: | ea8e0d49 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/4/) | 2020-08-24 08:53:11 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-085100-ea8e0d.zip) |
| :heavy_check_mark: | ea8e0d49 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/4/) | 2020-08-24 08:58:13 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-085056-ea8e0d.aar) |
| | | | | | | |
| :heavy_check_mark: | 5b215295 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/5/) | 2020-08-24 09:12:12 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-091043-5b2152.zip) |
| :heavy_check_mark: | 5b215295 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/5/) | 2020-08-24 09:12:39 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-091044-5b2152.zip) |
| :heavy_check_mark: | 5b215295 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/5/) | 2020-08-24 09:16:41 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-091041-5b2152.aar) |
| | | | | | | |
| :heavy_check_mark: | 265ef1fa | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/6/) | 2020-08-24 09:13:49 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-091214-265ef1.zip) |
| :heavy_check_mark: | 265ef1fa | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/6/) | 2020-08-24 09:14:27 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-091241-265ef1.zip) |
| :heavy_check_mark: | 265ef1fa | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/6/) | 2020-08-24 09:22:16 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-091643-265ef1.aar) |
| | | | | | | |
| :heavy_check_mark: | 92eef225 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/7/) | 2020-08-24 09:53:50 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-095213-92eef2.zip) |
| :heavy_check_mark: | 92eef225 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/7/) | 2020-08-24 09:55:44 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-095212-92eef2.zip) |
| :heavy_check_mark: | 92eef225 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/7/) | 2020-08-24 09:58:49 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-095210-92eef2.aar) |
| | | | | | | |
| :heavy_check_mark: | 16f5adec | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/8/) | 2020-08-24 10:19:05 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-101735-16f5ad.zip) |
| :heavy_check_mark: | 16f5adec | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/8/) | 2020-08-24 10:19:23 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-101736-16f5ad.zip) |
| :heavy_check_mark: | 16f5adec | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/8/) | 2020-08-24 10:24:54 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-101733-16f5ad.aar) |
| | | | | | | |
| :heavy_check_mark: | 00bdb306 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/9/) | 2020-08-24 11:05:17 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-110338-00bdb3.zip) |
| :heavy_check_mark: | 00bdb306 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/9/) | 2020-08-24 11:05:18 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-110340-00bdb3.zip) |
| :heavy_check_mark: | 00bdb306 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/9/) | 2020-08-24 11:09:57 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-110336-00bdb3.aar) |
| :heavy_check_mark: | 00bdb306 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/10/) | 2020-08-24 13:15:26 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-131351-00bdb3.zip) |
| :heavy_check_mark: | 00bdb306 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/10/) | 2020-08-24 13:16:02 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-131353-00bdb3.zip) |
| :heavy_check_mark: | 00bdb306 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/11/) | 2020-08-24 13:54:46 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-135319-00bdb3.zip) |
| :heavy_check_mark: | 00bdb306 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/11/) | 2020-08-24 13:55:48 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-135320-00bdb3.zip) |
| | | | | | | |
| :heavy_check_mark: | 4714cb3d | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/12/) | 2020-08-24 18:40:23 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-183851-4714cb.zip) |
| :heavy_check_mark: | 4714cb3d | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/12/) | 2020-08-24 18:40:45 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-183852-4714cb.zip) |
| :heavy_check_mark: | 4714cb3d | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/10/) | 2020-08-24 18:44:42 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-183849-4714cb.aar) |
| | | | | | | |
| :heavy_check_mark: | 441dcb52 | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/13/) | 2020-08-24 19:12:19 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200824-191046-441dcb.zip) |
| :heavy_check_mark: | 441dcb52 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/13/) | 2020-08-24 19:12:38 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200824-191048-441dcb.zip) |
| :heavy_check_mark: | 441dcb52 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/11/) | 2020-08-24 19:16:34 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200824-191044-441dcb.aar) |
| | | | | | | |
| :heavy_check_mark: | 9183b966 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/14/) | 2020-08-25 10:19:04 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200825-101716-9183b9.zip) |
| :heavy_check_mark: | 9183b966 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/14/) | 2020-08-25 10:19:12 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200825-101717-9183b9.zip) |
| :heavy_check_mark: | 9183b966 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/12/) | 2020-08-25 10:23:12 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200825-101714-9183b9.aar) |
| | | | | | | |
| :heavy_check_mark: | 5503dd0b | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/15/) | 2020-08-25 12:25:49 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200825-122412-5503dd.zip) |
| :heavy_check_mark: | 5503dd0b | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/15/) | 2020-08-25 12:26:00 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200825-122413-5503dd.zip) |
| :heavy_check_mark: | 5503dd0b | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/13/) | 2020-08-25 12:29:55 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200825-122409-5503dd.aar) |
| | | | | | | |
| :heavy_check_mark: | 4ff059b1 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/16/) | 2020-08-26 05:55:54 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200826-055422-4ff059.zip) |
| :heavy_check_mark: | 4ff059b1 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/16/) | 2020-08-26 05:56:23 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200826-055423-4ff059.zip) |
| :heavy_check_mark: | 4ff059b1 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/14/) | 2020-08-26 06:00:16 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200826-055420-4ff059.aar) |
| | | | | | | |
| :heavy_check_mark: | 8b7417d6 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/17/) | 2020-08-26 07:24:33 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200826-072305-8b7417.zip) |
| :heavy_check_mark: | 8b7417d6 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/17/) | 2020-08-26 07:24:40 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200826-072306-8b7417.zip) |
| :heavy_check_mark: | 8b7417d6 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/15/) | 2020-08-26 07:28:49 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200826-072303-8b7417.aar) |
| | | | | | | |
| :heavy_check_mark: | fc64fd72 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/18/) | 2020-08-26 07:37:25 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200826-073550-fc64fd.zip) |
| :heavy_check_mark: | fc64fd72 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/18/) | 2020-08-26 07:37:27 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200826-073552-fc64fd.zip) |
| :heavy_check_mark: | fc64fd72 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/16/) | 2020-08-26 07:40:58 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200826-073549-fc64fd.aar) |
| | | | | | | |
| :heavy_check_mark: | 00cd3982 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/19/) | 2020-08-26 08:43:12 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200826-084138-00cd39.zip) |
| :heavy_check_mark: | 00cd3982 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/19/) | 2020-08-26 08:43:33 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200826-084139-00cd39.zip) |
| :heavy_check_mark: | 00cd3982 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/17/) | 2020-08-26 08:47:38 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200826-084136-00cd39.aar) |
| | | | | | | |
| :heavy_check_mark: | afcae037 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/20/) | 2020-08-26 12:34:22 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200826-123251-afcae0.zip) |
| :heavy_check_mark: | afcae037 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/20/) | 2020-08-26 12:34:45 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200826-123253-afcae0.zip) |
| :heavy_check_mark: | afcae037 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/18/) | 2020-08-26 12:38:58 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200826-123249-afcae0.aar) |
| | | | | | | |
| :heavy_check_mark: | 9acdcefc | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/21/) | 2020-08-27 12:43:08 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200827-124125-9acdce.zip) |
| :heavy_check_mark: | 9acdcefc | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/21/) | 2020-08-27 12:43:25 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200827-124127-9acdce.zip) |
| :heavy_check_mark: | 9acdcefc | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/19/) | 2020-08-27 12:47:21 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200827-124124-9acdce.aar) |
| | | | | | | |
| :heavy_check_mark: | 51a5b168 | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/22/) | 2020-08-27 12:59:24 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200827-125747-51a5b1.zip) |
| :heavy_check_mark: | 51a5b168 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/22/) | 2020-08-27 12:59:33 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200827-125749-51a5b1.zip) |
| :heavy_check_mark: | 51a5b168 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/20/) | 2020-08-27 13:03:51 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200827-125745-51a5b1.aar) |
| | | | | | | |
| :heavy_check_mark: | 460dd468 | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/23/) | 2020-08-27 13:01:37 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200827-130002-460dd4.zip) |
| :heavy_check_mark: | 460dd468 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/23/) | 2020-08-27 13:01:47 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200827-130004-460dd4.zip) |
| :heavy_check_mark: | 460dd468 | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/21/) | 2020-08-27 13:09:44 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200827-130353-460dd4.aar) |
| | | | | | | |
| :heavy_check_mark: | 83d59916 | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/24/) | 2020-08-27 13:32:38 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200827-133108-83d599.zip) |
| :heavy_check_mark: | 83d59916 | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/24/) | 2020-08-27 13:32:57 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200827-133110-83d599.zip) |
| :heavy_check_mark: | 83d59916 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/22/) | 2020-08-27 13:37:04 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200827-133106-83d599.aar) |
| | | | | | | |
| :heavy_check_mark: | eb04c1ab | [#25](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/25/) | 2020-08-27 13:36:30 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200827-133451-eb04c1.zip) |
| :heavy_check_mark: | eb04c1ab | [#25](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/25/) | 2020-08-27 13:36:35 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200827-133453-eb04c1.zip) |
| :heavy_check_mark: | eb04c1ab | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/23/) | 2020-08-27 13:42:44 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200827-133706-eb04c1.aar) |
| | | | | | | |
| :heavy_check_mark: | d1c8676f | [#26](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/26/) | 2020-08-27 16:56:32 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200827-165454-d1c867.zip) |
| :heavy_check_mark: | d1c8676f | [#26](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/26/) | 2020-08-27 16:58:19 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200827-165452-d1c867.zip) |
| :heavy_check_mark: | d1c8676f | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/24/) | 2020-08-27 17:01:24 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200827-165451-d1c867.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7a1fdc9d | [#27](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/27/) | 2020-09-01 08:04:43 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200901-080251-7a1fdc.zip) |
| :heavy_check_mark: | 7a1fdc9d | [#27](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/27/) | 2020-09-01 08:04:52 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200901-080253-7a1fdc.zip) |
| :heavy_check_mark: | 7a1fdc9d | [#25](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/25/) | 2020-09-01 08:08:28 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200901-080249-7a1fdc.aar) |
| | | | | | | |
| :heavy_check_mark: | 4a87764c | [#28](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2022/28/) | 2020-09-01 08:33:14 | ~1 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200901-083132-4a8776.zip) |
| :heavy_check_mark: | 4a87764c | [#28](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2022/28/) | 2020-09-01 08:33:26 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200901-083134-4a8776.zip) |
| :heavy_check_mark: | 4a87764c | [#26](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2022/26/) | 2020-09-01 08:37:16 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200901-083130-4a8776.aar) |

cammellos committed:
shuffle servers and limit & filter by chat id

cammellos committed:
handle notifications at-most-once

cammellos committed:
Don't add contact on update

When sending a contact update we automatically added the contact,
but that resulted in the contact not being synced correctly as
`saveContact` will not trigger the side effects.

For now I have removed this behavior. Ideally we should have a single
call that handles the side effects, but for that ENS names should be
stored in messenger, so we can propagate it.

Samyoul commented on config/cli/fleet-eth.prod.json:
```diff
@@ -23,6 +23,17 @@
       "enode://8a64b3c349a2e0ef4a32ea49609ed6eb3364be1110253c20adc17a3cebbc39a219e5d3e13b151c0eee5d8e0f9a8ba2cd026014e67b41a4ab7d1d5dd67ca27427@178.128.142.94:443",
       "enode://44160e22e8b42bd32a06c1532165fa9e096eebedd7fa6d6e5f8bbef0440bc4a4591fe3651be68193a7ec029021cdb496cfe1d7f9f1dc69eb99226e6f39a7a5d4@35.225.221.245:443"
     ],
+    "PushNotificationsServers": [
+      "04606ae04a71e5db868a722c77a21c8244ae38f1bd6e81687cc6cfe88a3063fa1c245692232f64f45bd5408fed5133eab8ed78049332b04f9c110eac7f71c1b429",
+      "04c42f368a23fa98ee546fd247220759062323249ef657d26d357a777443aec04db1b29a3a22ef3e7c548e18493ddaf51a31b0aed6079bd6ebe5ae838fcfaf3a49",
+      "04ee2b53b0ace9692167a410514bca3024695dbf0e1a68e1dff9716da620efb195f04a4b9e873fb9b74ac84de801106c465b8e2b6c4f0d93b8749d1578bfcaf03e",
+      "042c8de3cbb27a3d30cbb5b3e003bc722b126f5aef82e2052aaef032ca94e0c7ad219e533ba88c70585ebd802de206693255335b100307645ab5170e88620d2a81",
+      "047aa648d6e855950b2e3d3bf220c496e0cae4adfddef3e1e6062e6b177aec93bc6cdcf1282cb40d1656932ebfdd565729da440368d7c4da7dbd4d004b1ac02bf8",
+      "0430211cbd81c25f07b03a0196d56e6ce4604bb13db773ff1c0ea2253547fafd6c06eae6ad3533e2ba39d59564cfbdbb5e2ce7c137a5ebb85e99dcfc7a75f99f55",
+      "04e85f1d4209f2f99da801af18db8716e584a28ad0bdc47fbdcd8f26af74dbd97fc279144680553ec7cd9092afe683ddea1e0f9fc571ebcb4b1d857c03a088853d",
+      "048a64b3c349a2e0ef4a32ea49609ed6eb3364be1110253c20adc17a3cebbc39a219e5d3e13b151c0eee5d8e0f9a8ba2cd026014e67b41a4ab7d1d5dd67ca27427",
+      "0444160e22e8b42bd32a06c1532165fa9e096eebedd7fa6d6e5f8bbef0440bc4a4591fe3651be68193a7ec029021cdb496cfe1d7f9f1dc69eb99226e6f39a7a5d4"
+    ],
```
Curious more than anything. These look like a list of public keys, how are they defined? Were they just randomly generated? Where are the corresponding private keys stored (if stored at all)?

cammellos replied:
PN servers will run the same as mailservers, so these are the public keys of the mailservers (you can check the enode://.. above, and they should be the same)

jakubgs replied:
Why are these just public keys and not full enodes?

cammellos committed:
Handle connection change

Currently messenger has no notion of being online.
This might cause a problem as we retry to register with a push
notification server even if not connected to any peer, which will
inevitably fail.
This commit adds a method `handleConnectionChange` that will be called
every time the connection change state.

Samyoul commented on protocol/messenger.go:
```diff
@@ -78,6 +78,7 @@ type Messenger struct {
 	installationID             string
 	mailserver                 []byte
 	database                   *sql.DB
+	quit                       chan struct{}
```
Why is this a `struct{]` and not a bool for example?

Does the `quit` field have a write event? I can't see it?

cammellos replied:
not sure, I always seen that being used, I think it's because of memory https://stackoverflow.com/questions/52035390/why-using-chan-struct-when-wait-something-done-not-chan-interface

cammellos replied:
it's closed in `Shutdown()` https://github.com/status-im/status-go/pull/2022/files#diff-5df3979bc10f2ff16f7942ec557947ecR570

Samyoul replied:
That's interesting. It feels hacky, but I see why one would use `struct{}`.

Samyoul commented on protocol/pushnotificationclient/client.go:
```diff
@@ -287,15 +314,47 @@ func (c *Client) Reregister(contactIDs []*ecdsa.PublicKey, mutedChatIDs []string
+	// shuffle and pick n at random
+	shuffledServers := make([]*ecdsa.PublicKey, len(servers))
+	copy(shuffledServers, c.config.DefaultServers)
+	mrand.Shuffle(len(shuffledServers), func(i, j int) {
+		shuffledServers[i], shuffledServers[j] = shuffledServers[j], shuffledServers[i]
+	})
```
`mrand` doesn't seem to have its seed set. This will default to the same seed each time.

cammellos replied:
ah didn't know, good point!

Samyoul replied:
It's a quirk of go, in go the random libraries aren't random by default.

Samyoul reviewed (approved):
Lot's of good work, the only point of concern is setting the `mrand` seed

cammellos committed:
seed mrand

cammellos commented:
because push notifications servers are status clients, same as a status any
other client, it s not through devp2p, so you communicate with them through
waku and not directly

On Thu, Aug 27, 2020, 20:00 Jakub <notifications@github.com> wrote:

> *@jakubgs* commented on this pull request.
> ------------------------------
>
> In config/cli/fleet-eth.prod.json
> <https://github.com/status-im/status-go/pull/2022#discussion_r478599272>:
>
> > +    "PushNotificationsServers": [
> +      "04606ae04a71e5db868a722c77a21c8244ae38f1bd6e81687cc6cfe88a3063fa1c245692232f64f45bd5408fed5133eab8ed78049332b04f9c110eac7f71c1b429",
> +      "04c42f368a23fa98ee546fd247220759062323249ef657d26d357a777443aec04db1b29a3a22ef3e7c548e18493ddaf51a31b0aed6079bd6ebe5ae838fcfaf3a49",
> +      "04ee2b53b0ace9692167a410514bca3024695dbf0e1a68e1dff9716da620efb195f04a4b9e873fb9b74ac84de801106c465b8e2b6c4f0d93b8749d1578bfcaf03e",
> +      "042c8de3cbb27a3d30cbb5b3e003bc722b126f5aef82e2052aaef032ca94e0c7ad219e533ba88c70585ebd802de206693255335b100307645ab5170e88620d2a81",
> +      "047aa648d6e855950b2e3d3bf220c496e0cae4adfddef3e1e6062e6b177aec93bc6cdcf1282cb40d1656932ebfdd565729da440368d7c4da7dbd4d004b1ac02bf8",
> +      "0430211cbd81c25f07b03a0196d56e6ce4604bb13db773ff1c0ea2253547fafd6c06eae6ad3533e2ba39d59564cfbdbb5e2ce7c137a5ebb85e99dcfc7a75f99f55",
> +      "04e85f1d4209f2f99da801af18db8716e584a28ad0bdc47fbdcd8f26af74dbd97fc279144680553ec7cd9092afe683ddea1e0f9fc571ebcb4b1d857c03a088853d",
> +      "048a64b3c349a2e0ef4a32ea49609ed6eb3364be1110253c20adc17a3cebbc39a219e5d3e13b151c0eee5d8e0f9a8ba2cd026014e67b41a4ab7d1d5dd67ca27427",
> +      "0444160e22e8b42bd32a06c1532165fa9e096eebedd7fa6d6e5f8bbef0440bc4a4591fe3651be68193a7ec029021cdb496cfe1d7f9f1dc69eb99226e6f39a7a5d4"
> +    ],
>
> Why are these just public keys and not full enodes?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/status-im/status-go/pull/2022#discussion_r478599272>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHYJMDVX4KEB4WS3O3V2HTSC2NMDANCNFSM4QF2NQ6Q>
> .
>

jakubgs commented on protocol/pushnotificationclient/client.go:
```diff
@@ -60,13 +60,21 @@ const pushNotificationBackoffTime int64 = 2
+type ServerType int
+
+const (
+	ServerTypeDefault = iota + 1
```
This is some black magic.

Samyoul replied:
iota is one of my favourite features of go.
